awsmp/models.py

```diff
@@ -808,6 +853,9 @@ def get_entity_from_yaml(yaml_config: dict[str, Any]) -> EntityModel:
         :return: An instance of `EntityModel` create from the yaml_config
         :rtype: EntityModel
         """
+
+        yaml_config = EntityModel._mask_unchecked_fields(yaml_config)
```

Comment:
should we do this in `cli.py` instead? (or perhaps introduce a `diff_entity` public api method in `_driver.py`) conceivably we could want to use `get_entity_from_yaml` outside the diff context. `_mask_unchecked_fields` would then overwrite potentially valid values